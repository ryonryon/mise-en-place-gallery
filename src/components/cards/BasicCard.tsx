import styled from "@emotion/styled";
import * as React from "react";

interface Props {
  children: React.ReactNode;
  width: string;
  height: string;
}

const BasicCard = ({ children, width, height }: Props) => {
  const StyledCard = styled.div`
    display: inline-block;
    width: ${width};
    ${height ? `height: ${height};` : ""}
    text-align: center;
    box-shadow: 1px 1px 4px black;
    margin: 16px;

    * {
      padding: 8px 16px;
    }

    &: hover {
      box-shadow: 2px 2px 8px black;
    }
  `;

  return <StyledCard>{children}</StyledCard>;
};

BasicCard.defaultProps = {
  width: "100%",
  height: null
};

export default BasicCard;
