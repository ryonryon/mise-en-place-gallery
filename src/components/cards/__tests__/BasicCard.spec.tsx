import * as React from "react";
import * as renderer from "react-test-renderer";

import BasicCard from "../BasicCard";

describe("BasicCard test", () => {
  it("tests BasicCard renders correctly ", () => {
    const basicCard = renderer
      .create(
        <BasicCard>
          <h1>test</h1>
          <p>test test</p>
        </BasicCard>
      )
      .toJSON();
    expect(basicCard).toMatchSnapshot();
  });
});
