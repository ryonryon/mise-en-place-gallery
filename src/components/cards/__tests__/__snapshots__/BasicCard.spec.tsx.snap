// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicCard test tests BasicCard renders correctly  1`] = `
<div
  className="css-1pcjs3q"
>
  <h1>
    test
  </h1>
  <p>
    test test
  </p>
</div>
`;
